### DISCUSSION TITLE
Resize a GIF to a specific file size constraint

### DISCUSSION THREAD

**Discussion by alexkozler** [SixLabors/ImageSharp#1340] [Q&A] [answered]
I have a 640x640 GIF that is 724KB. My goal is to have this be below 600KB but I don't want to have to do a trial-and-error loop of resizing the dimensions or compression of the GIF to reach the desired max file size. I am POSTing image data to a third party API and they don't care about the file type or dimensions, but the file size limit is strict.

Is there an optimal way to get an image to meet a file size limitation on-the-fly rather than repeatedly shrinking/compressing it until it reaches the desired constraints?

**Comment by JimBobSquarePants** [comment, marked as answer]
This question has come up a few times. Here's the answer (I'm afraid you're not going to like it).

https://github.com/SixLabors/ImageSharp/discussions/1305#discussioncomment-47054
